src/services/useDisplayConfig/index.ts

```diff
@@ -117,24 +153,25 @@ export function useDisplayConfig(
 export function useDisplayConfigFilter(
   baseUrl: string,
   filter: MaybeRefOrGetter<Filter>,
+  startTime: MaybeRefOrGetter<Date | undefined>,
+  endTime: MaybeRefOrGetter<Date | undefined>,
 ): UseDisplayConfigReturn {
   const piProvider = new PiWebserviceProvider(baseUrl, {
     transformRequestFn: createTransformRequestFn(),
   })
 
   const displayConfig = ref<DisplayConfig>()
   const displays = ref<DisplayConfig[]>()
+  const response = ref<ActionsResponse>()
 
   watchEffect(async () => {
     const _filter = toValue(filter)
-    let response: ActionsResponse
-    let nodeId = undefined
     if (isFilterActionsFilter(_filter)) {
       if (!_filter.filterId) return
-      response = await piProvider.getFilterActions(_filter)
+      response.value = await piProvider.getFilterActions(_filter)
     } else if (isTimeSeriesGridActionsFilter(_filter)) {
-      response = await piProvider.getTimeSeriesGridActions(_filter)
-      response.results.forEach((result) => {
+      response.value = await piProvider.getTimeSeriesGridActions(_filter)
+      response.value.results.forEach((result) => {
```

Comment:
Not your code, and I haven't looked in too much detail, but could this just be `flatMap`?